features/steps/test_behave.py

```diff
@@ -1,108 +1,229 @@
 #!/usr/bin/python -tt
 
 from behave import *
-import os, sys, subprocess, glob
+import os
+import subprocess
+import glob
+import re
 
 DNF_FLAGS = ['-y', '--disablerepo=*', '--nogpgcheck']
 RPM_INSTALL_FLAGS = ['-Uvh']
 RPM_ERASE_FLAGS = ['-e']
 
+
 def _left_decorator(item):
-  " Removed packages "
-  return u'-' + item
+    """ Removed packages """
+    return u'-' + item
+
 
 def _right_decorator(item):
-  " Installed packages "
-  return u'+' + item
+    """ Installed packages """
+    return u'+' + item
+
 
 def find_pkg(pkg):
-  " Find the package file in the repository "
-  candidates = glob.glob('/repo/'+pkg+'*.rpm')
-  if len(candidates) == 0:
-    print("No candidates for: '{0}'".format(pkg))
-  assert len(candidates) == 1
-  return candidates[0]
+    """ Find the package file in the repository """
+    candidates = glob.glob('/repo/' + pkg + '*.rpm')
+    if len(candidates) == 0:
+        print("No candidates for: '{0}'".format(pkg))
+    assert len(candidates) == 1
+    return candidates[0]
+
 
 def decorate_rpm_packages(pkgs):
-  " Converts package names like TestA, TestB into absolute paths "
-  return [find_pkg(p) for p in pkgs]
+    """ Converts package names like TestA, TestB into absolute paths """
+    return [find_pkg(p) for p in pkgs]
+
 
 def get_package_list():
-  " Gets all installed packages in the system "
-  pkgstr = subprocess.check_output(['rpm', '-qa', '--queryformat', '%{NAME}\n'])
-  return pkgstr.splitlines()
+    """ Gets all installed packages in the system """
+    pkgstr = subprocess.check_output(['rpm', '-qa', '--queryformat', '%{NAME}\n'])
+    return pkgstr.splitlines()
+
+
+def get_package_version_list():
+    """ Gets all installed packages in the system with version"""
+    pkgverstr = subprocess.check_output(['rpm', '-qa', '--queryformat', '%{NAME}-%{VERSION}-%{RELEASE}\n'])
+    return pkgverstr.splitlines()
+
+
+def get_package_dnf_version_list():
+    """ Gets all installed packages in the system with version to check that dnf has same data about installed packages"""
+    pkgverstr = subprocess.check_output(['dnf', 'repoquery', '--installed', '-C', '--queryformat', '%{name}.%{version}.%{release}\n'])
+    pkgverstr = pkgverstr.splitlines()
+    return pkgverstr[1:]
```

Comment:
this looks hacky (probably because of metadata information), add `-q` to the command instead, please